app/models/bank_transaction.rb

```diff
@@ -33,4 +33,55 @@ class BankTransaction < ActiveRecord::Base
   def image_url
     'data:image/png;base64,' + Base64.encode64(self.image)
   end
+
+  def assign_to_invoice
+    return false unless supplier
+
+    content = text
+    content += "\n" + reference if reference.present?
+    invoices = supplier.invoices.unpaid.select {|i| content.include? i.number}
+    invoices_sum = invoices.inject(0) {|sum,i| sum + i.amount }
+    return false if amount != -invoices_sum
```

Comment:
Just to check: may this introduce float comparison problems?

Reply:
i don't see any problems since only add values